File: src/main/views/@default/cache/cleanPolicy.js
Tea.context(function () {
	this.isCleaning = false;
	this.cleanResult = "";
	this.clean = function () {
		if (!window.confirm("确定要清理此策略关联的所有缓存吗？")) {
			return;
		}
		this.isCleaning = true;
		this.cleanResult = "";

		this.$post(".cleanPolicy")
			.params({
				"filename": this.policy.filename
			})
			.success(function (resp) {
				this.cleanResult = resp.data.result;
			})
			.fail(function () {
				this.cleanResult = resp.data.result;
			})
			.done(function () {
				this.isCleaning = false;
			})
	};
});

File: internal/teaweb/actions/default/cache/cleanPolicy.go
package cache

import (
	"github.com/TeaWeb/build/internal/teacache"
	"github.com/TeaWeb/build/internal/teacluster"
	"github.com/TeaWeb/build/internal/teaconfigs/shared"
	"github.com/TeaWeb/build/internal/teaweb/actions/default/actionutils"
	"github.com/iwind/TeaGo/maps"
)

type CleanPolicyAction struct {
	actionutils.ParentAction
}

// 清理
func (this *CleanPolicyAction) Run(params struct {
	Filename string
}) {
	this.SecondMenu("clean")

	policy := shared.NewCachePolicyFromFile(params.Filename)
	if policy == nil {
		this.Data["result"] = "找不到Policy"
		this.Fail()
	}
	this.Data["policy"] = policy
	this.Show()
}

// 执行清理
func (this *CleanPolicyAction) RunPost(params struct {
	Filename string
}) {
	policy := shared.NewCachePolicyFromFile(params.Filename)
	if policy == nil {
		this.Data["result"] = "找不到Policy"
		this.Fail()
	}

	manager := teacache.FindCachePolicyManager(params.Filename)
	if manager == nil {
		manager = teacache.NewManagerFromConfig(policy)
		defer func() {
			_ = manager.Close()
		}()
	}

	if manager == nil {
		this.Fail("找不到管理器")
	}

	err := manager.Clean()
	if err != nil {
		this.Data["result"] = err.Error()
		this.Fail()
	}

	// 同步到集群
	action := new(teacluster.RunAction)
	action.Cmd = "cache.clean"
	action.Data = maps.Map{
		"filename": params.Filename,
	}
	err = teacluster.SharedManager.Write(action)
	if err != nil {
		this.Fail("同步到集群失败：" + err.Error())
	}

	this.Data["result"] = "清理完成"
	this.Success()
}
